modules/reindex/src/main/java/org/opensearch/index/reindex/remote/RemoteVersion.java

```diff
@@ -0,0 +1,162 @@
+    public static final RemoteVersion V_2_3_3 = new RemoteVersion(2, 3, 3, null);
+    public static final RemoteVersion V_5_0_0 = new RemoteVersion(5, 0, 0, null);
+    public static final RemoteVersion V_6_0_0 = new RemoteVersion(6, 0, 0, null);
+    public static final RemoteVersion V_6_3_0 = new RemoteVersion(6, 3, 0, null);
+    public static final RemoteVersion V_7_0_0 = new RemoteVersion(7, 0, 0, null);
+
+    // OpenSearch versions
+    public static final RemoteVersion OPENSEARCH_1_0_0 = new RemoteVersion(1, 0, 0, "opensearch");
+    public static final RemoteVersion OPENSEARCH_2_0_0 = new RemoteVersion(2, 0, 0, "opensearch");
+    public static final RemoteVersion OPENSEARCH_3_1_0 = new RemoteVersion(3, 1, 0, "opensearch");
+
+    public RemoteVersion(int major, int minor, int revision, String distribution) {
+        this.major = major;
+        this.minor = minor;
+        this.revision = revision;
+        this.distribution = distribution != null ? distribution.toLowerCase(Locale.ROOT) : "elasticsearch";
+    }
+
+    /**
+     * Parse version string like "7.10.2" or "1.0.0"
+     */
+    public static RemoteVersion fromString(String version, String distribution) {
+        if (version == null || version.trim().isEmpty()) {
+            throw new IllegalArgumentException("Version string cannot be null or empty");
+        }
+
+        // Remove snapshot and pre-release qualifiers
+        String cleanVersion = version.replace("-SNAPSHOT", "").replaceFirst("-(alpha\\d+|beta\\d+|rc\\d+)", "");
+
+        String[] parts = cleanVersion.split("\\.");
+        if (parts.length < 2) {
+            throw new IllegalArgumentException("Invalid version format: " + version);
+        }
+
+        try {
+            int major = Integer.parseInt(parts[0]);
+            int minor = Integer.parseInt(parts[1]);
+            int revision = parts.length > 2 ? Integer.parseInt(parts[2]) : 0;
+
+            return new RemoteVersion(major, minor, revision, distribution);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Invalid version format: " + version, e);
+        }
+    }
+
+    public static RemoteVersion fromString(String version) {
+        return fromString(version, "elasticsearch");
+    }
+
+    public boolean before(RemoteVersion other) {
+        return this.compareTo(other) < 0;
+    }
+
+    public boolean onOrAfter(RemoteVersion other) {
+        return this.compareTo(other) >= 0;
+    }
+
+    public boolean onOrBefore(RemoteVersion other) {
+        return this.compareTo(other) <= 0;
+    }
+
+    public boolean after(RemoteVersion other) {
+        return this.compareTo(other) > 0;
+    }
+
+    public boolean isOpenSearch() {
+        return "opensearch".equals(distribution);
+    }
+
+    public boolean isElasticsearch() {
+        return "elasticsearch".equals(distribution);
+    }
+
+    @Override
+    public int compareTo(RemoteVersion other) {
```

Comment:
For comparison purposes, I would say that OpenSearch is always after Elasticsearch. At least that should satisfy all of the checks in the code.